src/Symfony/Component/BrowserKit/AbstractBrowser.php

```diff
@@ -256,27 +256,28 @@ public function getRequest(): object
     /**
      * Clicks on a given link.
      */
-    public function click(Link $link): Crawler
+    public function click(Link $link, array $serverParameters = []): Crawler
```

Comment:
@syl20b do you have any use case for it? Because for me it's quite going again the component. BrowserKit is made to simulate Browser interaction, and a user using a browser never touch those server parameters.

Reply:
The purpose of this, is to be able to specify some HTTP headers that must be "persitent" during the scenario (like SSO headers for example). 
This is usefull when using the Symfony Client, which execute the request without running a real browser.

Reply:
Would it be better if we can set these headers only once before making several HTTP calls?

Reply:
Server parameters can be set in the AbstractBrowser::__construct() method, but for now, those parameters are persistent accross testing scenario, and never reset (CF #48839). 

Moreover, this is a flexible way to handle headers through the scenario.
And finally, it's consistent with the AbstractBrowser::submit() and AbstractBrowser::submitForm() methods which allow such a feature.

Reply:
I tend to agree with @alexislefebvre here.
/cc @symfony/mergers

Reply:
On the one hand I agree with @alexislefebvre but as @syl20b points out, it would make it more consistent with the form methods.

Thinking they should be consistent - one way or the other. But should click have a method param? (I would actually have a use case for this).

Reply:
@fabpot @kbond 
I agree that consistency is the most important thing. 
I can rework it, and deprecate form methods before removing their parameters in the next major release. But doing that way we need a way to reset server parameters between each scenario, wich is not the case actually (CF #48839). 
What do you think ?

Reply:
What about a new method to set server parameter once for all, but only for the current scenario?

Reply:
@lyrixx Won’t that be too confusing with the current `setServerParameters()`  method ?

Reply:
I forgot this one. Why `setServerParameters` is not enough?